### ISSUE TITLE
Get rid of the .htaccess generated security

### ISSUE THREAD

**Issue by amtriorix** [PrestaShop/PrestaShop#12156]
There are a lot, a lot security issues with the .htaccess in a multistore environment.
Not to mention it is generated by prestashop and the running www user.

Using .htaccess is a bad practice. It is one of the main reasons why Nginx does not support .htaccess.

The reason is obvious. When a script is compromised, it is possible to change the security settings of
.htaccess as well, especially when this is generated by php scripts.

Get rid of that .htaccess file. Provide better methods so we can hardcode the settings in de server
settings, or provide an option to have the two options, configuration .htaccess less or with .htaccess.

Additionally, in a multistore, the .htaccess does fill in a lot of duplicate rules in regards of multisites.
The rules are not tuned and does not use properly regex to avoid too much redirection formula.

When our site goes 500, well it is mainly due to the corrupt .htaccess generated by prestashop scripts.
or in worse case, hacked.

As said in previous tickets, more attention should go to security  like:
1) .htaccess and best to exclude it
2) the umask and chmod of files, there is NO NEED to provide the user OTHER, permissions of any kind
3) to limit writable directories and files
4) to tune scripts so they work properly with PHP-FPM and more secure settings as the general www
    or (www-data) user for webserver and php application server.
5) To provide an install mode that can be tweaked in the config file, then set this to run mode,
    where much less permissions are allowed. (See for this the system used in Xoops) As once the
    site is running, there is no need to give permissions to install/modify/delete modules etc.

For this reason I call for a break and to focus on security in next minor versions

**Comment by PierreRambaud** [comment]
Hi and thanks for the feedback,

1) I totally agree about the `.htaccess`, it is a bad practice, and this is why Apache allows you to disable it (http://httpd.apache.org/docs/2.4/mod/core.html#allowoverride). But don't forget, many websites that are running under PrestaShop are not hosted in a dedicated server and can't configure Apache.

2) We are planning to remove the umask force, there is already a pull request about that.

3) We are in a legacy context and we can not break everything, but of course, we agree, we need to reduce the writable directories. But this is not a priority.

4) Can you tell us more? I run PrestaShop under Nginx and Apache with specific user, and sometimes with www-data and I do not have any problem :thinking: 

5) Can you please describe more what you mean ? I am not sure I understand it well, especially the difference between install and usage.

We will do our best to help you and find solutions.

Kind regards,

**Comment by khouloudbelguith** [comment]
Hi @amtriorix,

If we had no news from you for more than 30 days, we will close this ticket.
Any news about your issue?
@PierreRambaud is waiting for your responses.
Thanks for your understanding!

**Comment by khouloudbelguith** [comment]
Hi @amtriorix,

Since we had no news from you for more than 30 days, I'll close this ticket. Feel free to open another one if you can give specific details.

Thanks!

**Comment by amtriorix** [comment]
Look, it is NOT MY DUTY to patch your software. I do test and report issues.

**Comment by khouloudbelguith** [comment]
Hi @amtriorix,

Sorry, but @PierreRambaud was waiting for your response.
We don't have a response for more than 30days.
Thanks!

**Comment by matks** [comment]
> Look, it is NOT MY DUTY to patch your software. I do test and report issues.

No one said so 😄we were just interested in hearing more about your usecase.

Unfortunately, as @PierreRambaud explained, we cannot make such a big change for minor versions. Better .htaccess management (and we **have** to keep it, some people have no other choice) and better directories management is scheduled, but not until next major version, PS 8.
